test/cpp/api/modules.cpp

```diff
@@ -3044,6 +3050,379 @@ TEST_F(ModulesTest, BCEWithLogitsLoss) {
   }
 }
 
+namespace detail {
+
+  namespace F = torch::nn::functional;
+
+  torch::Tensor _batchmatmul(const torch::Tensor& a, const torch::Tensor& b) {
+    assert(a.size(0) == b.size(0));
```

Comment:
should we use TORCH_INTERNAL_ASSERT instead?
``TORCH_INTERNAL_ASSERT(a.size(0) == b.size(0));``